### PR TITLE
Added support of XML for get single payment status

### PR DESCRIPTION
**Pull request by n0rdy**

### CODE REVIEW

File: xs2a-gateway-service-api-adapter/src/main/java/de/adorsys/xs2a/gateway/adapter/AbstractService.java [hunk 1]
```diff
@@ -25,16 +25,22 @@
...
+import java.io.ByteArrayOutputStream;
 import java.io.IOException;
 import java.io.PushbackInputStream;
 import java.io.UncheckedIOException;
+import java.nio.charset.StandardCharsets;
 import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 public abstract class AbstractService {
```
Comment (spetrychenko) [inline xs2a-gateway-service-api-adapter/src/main/java/de/adorsys/xs2a/gateway/adapter/AbstractService.java]: I propose to add logger and log all exception and all necessary actions

File: xs2a-gateway-service-api-adapter/src/main/java/de/adorsys/xs2a/gateway/adapter/AbstractService.java [hunk 2]
```diff
@@ -71,6 +77,34 @@ public void setHttpClient(HttpClient httpClient) {
...
+                try (ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
+                    byte[] buffer = new byte[1024];
+                    int length;
+                    while ((length = responseBody.read(buffer)) != -1) {
+                        baos.write(buffer, 0, length);
+                    }
+
+                    Matcher matcher = CHARSET_PATTERN.matcher(responseHeaders.getHeader(CONTENT_TYPE_HEADER));
+                    if (matcher.find()) {
+                        return baos.toString(matcher.group(1));
```
Comment (spetrychenko) [inline xs2a-gateway-service-api-adapter/src/main/java/de/adorsys/xs2a/gateway/adapter/AbstractService.java]: log defined charset

File: xs2a-gateway-service-api-adapter/src/main/java/de/adorsys/xs2a/gateway/adapter/AbstractService.java (hunk 1 above)
Comment (n0rdy) [reply xs2a-gateway-service-api-adapter/src/main/java/de/adorsys/xs2a/gateway/adapter/AbstractService.java]: Done, thanks. Please, take a look, whether it is done as you expected.

File: xs2a-gateway-service-api-adapter/src/main/java/de/adorsys/xs2a/gateway/adapter/AbstractService.java (hunk 2 above)
Comment (n0rdy) [reply xs2a-gateway-service-api-adapter/src/main/java/de/adorsys/xs2a/gateway/adapter/AbstractService.java]: Done, thanks!

File: xs2a-gateway-service-api-adapter/src/main/java/de/adorsys/xs2a/gateway/adapter/AbstractService.java [hunk 3]
```diff
@@ -92,16 +103,28 @@ public void setHttpClient(HttpClient httpClient) {
...
+                    String charset = null;
+
                     if (matcher.find()) {
-                        return baos.toString(matcher.group(1));
+                        charset = matcher.group(1);
                     }
 
-                    return baos.toString(StandardCharsets.UTF_8.name());
+                    if (charset == null) {
+                        charset = StandardCharsets.UTF_8.name();
```
Comment (spetrychenko) [inline xs2a-gateway-service-api-adapter/src/main/java/de/adorsys/xs2a/gateway/adapter/AbstractService.java]: please assign StandardCharsets.UTF_8.name() directly to the charset variable at line #107, after you can remove this unnecessary null check

File: xs2a-gateway-service-api-adapter/src/main/java/de/adorsys/xs2a/gateway/adapter/AbstractService.java (hunk 1 above)
Comment (spetrychenko) [reply xs2a-gateway-service-api-adapter/src/main/java/de/adorsys/xs2a/gateway/adapter/AbstractService.java]: Please rename LOG variable to lower case

File: xs2a-gateway-service-api-adapter/src/main/java/de/adorsys/xs2a/gateway/adapter/AbstractService.java (hunk 3 above)
Comment (n0rdy) [reply xs2a-gateway-service-api-adapter/src/main/java/de/adorsys/xs2a/gateway/adapter/AbstractService.java]: Done, thanks!

File: xs2a-gateway-service-api-adapter/src/main/java/de/adorsys/xs2a/gateway/adapter/AbstractService.java (hunk 1 above)
Comment (n0rdy) [reply xs2a-gateway-service-api-adapter/src/main/java/de/adorsys/xs2a/gateway/adapter/AbstractService.java]: Done, thanks!

**Pull request merged**